### PR TITLE
tests/main: document more tests

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Review by sergiocazzolato — APPROVED** [review]
Thanks

File: tests/main/parallel-install-common-dirs-undo/task.yaml [hunk 1]
```diff
@@ -1,5 +1,19 @@
 summary: Checks handling of common snap directories of parallel installed snaps
 
+details: |
+    Snapd allows installation of the same snap more than once by combining the
+    same snap name with different values of an instance key.
+
+    Installation of snaps under an instance key is somewhat different than
```
Comment (olivercalder) [inline tests/main/parallel-install-common-dirs-undo/task.yaml:7]: nitpick (for consistency with following line): `s/snaps/a snap/`

File: tests/main/parallel-install-common-dirs/task.yaml [hunk 2]
```diff
@@ -1,5 +1,18 @@
 summary: Checks handling of common snap directories of parallel installed snaps
 
+details: |
+    Snapd allows installation of the same snap more than once by combining the
+    same snap name with different values of an instance key.
+
+    Installation of snaps under an instance key is somewhat different than
```
Comment (olivercalder) [inline tests/main/parallel-install-common-dirs/task.yaml:7]: Same

File: tests/main/parallel-install-desktop/task.yaml [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+    One aspect of the instance system is that applications with desktop files
+    need a mechanism to alter each desktop file so that each instance can be
+    individually found and started in the desktop shell.
+
+    The test installs a snap with a desktop file and looks at the contents of
+    the instance-aware desktop file written by snapd, to ensure that in each
+    case the launched application is correct, and that some attributes contain
+    the name of the instance.
+
+    The test also checks that removal on one instance does not affect the
```
Comment (olivercalder) [inline tests/main/parallel-install-desktop/task.yaml:16]: `s/on /of /`?

File: tests/main/prefer/task.yaml [hunk 4]
```diff
@@ -1,5 +1,13 @@
 summary: Simple snap prefer test
 
+details: |
+    Snapd allows selection of preferred source of aliases for snap applications.
+    The "snap prefer" command can override manually created aliases as well as
+    default aliases of other snaps.
+
+    The test installs a snap with automatically enabled aliases, uses the "snap unalias" command
+    to remove them, then
```
Comment (olivercalder) [inline tests/main/prefer/task.yaml:9]: Unexpected EOF :)

**Review by olivercalder — COMMENTED** [review]
Thanks! Spotted a few small things

File: tests/main/prefer/task.yaml (hunk 4 above)
Comment (zyga) [reply tests/main/prefer/task.yaml:9]: Oh, what has happened here. Did I get out of insert mode and nuke some of the content while writing a commit message. Thank you for spotting this.

**Comment by zyga** [comment]
I'll pick this up now.

File: tests/main/prefer/task.yaml (hunk 4 above)
Comment (zyga) [reply tests/main/prefer/task.yaml]: Fixed

**Pull request merged**